container-search/src/main/java/com/yahoo/prelude/cluster/ClusterSearcher.java

```diff
@@ -228,8 +234,23 @@ private Result doSearch(Searcher searcher, Query query, Execution execution) {
         } else {
             String docType = schemas.iterator().next();
             query.getModel().setRestrict(docType);
-            return searcher.search(query, execution);
+            return perSchemaSearch(searcher, query, execution);
+        }
+    }
+
+    private Result perSchemaSearch(Searcher searcher, Query query, Execution execution) {
+        Set<String> restrict = query.getModel().getRestrict();
+        if (restrict.size() != 1) {
+            throw new IllegalStateException("perSchemaSearch must always be called with 1 schema, got: " + restrict.size());
+        }
+        for (String schema : restrict) {
+            Result result = searcher.search(query, execution);
```

Comment:
here it's invoked exactly 1 time, that's checked immediately previous.  I could use restrict.iterator().next() instead of a loop, to make it clearer that it only happens once?

Reply:
You're right; my bad. I didn't read the preceding lines carefully enough 🤦 I agree with your proposal.